returnn/tf/layers/basic.py

```diff
@@ -1341,6 +1341,17 @@ def __init__(self, position, axis, **kwargs):
     # (BatchAxes.., InputAxesBeforeGatherAxis, PositionAxes.., InputAxesAfterGatherAxis..)
     self.output.placeholder = tf.gather(params=params, indices=indices, axis=gather_axis, batch_dims=batch_dims)
 
+    if input_data.dim_tags[old_gather_axis].is_batch_dim():
+      for axis in self.output.size_placeholder:
+        new_size = tf.gather(params=self.output.size_placeholder[axis], indices=position_data.placeholder)
+        from ..util.data import Dim
+        Dim(
+          kind=Dim.Types.Spatial, description="%s_gather_axis" % self.name,
+          dyn_size=new_size, batch=self.output.batch,
+          src_data=self.output, src_axis=axis, auto_generated=True)
+        self.output.size_placeholder[axis] = new_size
```

Comment:
You don't use the `Dim` object you created?
Instead of assigning `size_placeholder`, I think it would be better to set the newly created dim tag.

Reply:
What is the usual way to set dim tags? I can't just reassign `self.output.dim_tags`. `declare_same_as` is used elsewhere, but not sure if it applies here.

Reply:
See most other layers. Usually you set `dim_tags` in `get_out_data_from_opts`. You should not assign a new dim tag in `__init__`. In `__init__`, you just might to assign the `dyn_size_ext` or `dyn_size_ext.placeholder` of a dim tag which was previously newly created in `get_out_data_from_opts`.